hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/recon/TestRecon.java

```diff
@@ -0,0 +1,385 @@
+        ReconUtils.class);
+
+    // initialize HTTPClient
+    httpClient = HttpClientBuilder
+        .create()
+        .setDefaultRequestConfig(config)
+        .build();
+
+    pauseInterval = conf.getTimeDuration(
+        conf.get(RECON_OM_SNAPSHOT_TASK_INTERVAL),
+        conf.get(RECON_OM_SNAPSHOT_TASK_INTERVAL_DEFAULT),
+        TimeUnit.MILLISECONDS);
+  }
+
+  @AfterClass
+  public static void shutdown() {
+    if (cluster != null) {
+      cluster.shutdown();
+    }
+  }
+
+  /**
+   * Returns a {@link CloseableHttpClient} configured by given configuration.
+   * If conf is null, returns a default instance.
+   *
+   * @param url        URL
+   * @return a JSON String Response.
+   */
+  private String makeHttpCall(String url)
+      throws IOException {
+    HttpGet httpGet = new HttpGet(url);
+    HttpResponse response = httpClient.execute(httpGet);
+    int errorCode = response.getStatusLine().getStatusCode();
+    HttpEntity entity = response.getEntity();
+
+    if ((errorCode == HTTP_OK) || (errorCode == HTTP_CREATED)) {
+      return EntityUtils.toString(entity);
+    }
+
+    if (entity != null) {
+      throw new IOException("Unexpected exception when trying to reach Ozone " +
+          "Manager, " + EntityUtils.toString(entity));
+    } else {
+      throw new IOException("Unexpected null in http payload," +
+          " while processing request");
+    }
+  }
+
+  @Test
+  public void testReconServer() throws Exception {
+    //add a vol, bucket and key
+    addKeys(0, 1);
+
+    //check if OM metadata has vol0/bucket0/key0 info
+    String ozoneKey = metadataManager.getOzoneKey(
+        "vol0", "bucket0", "key0");
+    OmKeyInfo keyInfo1 = metadataManager.getKeyTable().get(ozoneKey);
+
+    TableIterator<String, ? extends Table.KeyValue<String, OmKeyInfo>>
+        omKeyValueTableIterator = metadataManager.getKeyTable().iterator();
+
+    long omMetadataKeyCount = getTableKeyCount(omKeyValueTableIterator);
+
+    //verify if OM has /vol0/bucket0/key0
+    Assert.assertEquals("vol0", keyInfo1.getVolumeName());
+    Assert.assertEquals("bucket0", keyInfo1.getBucketName());
+
+    //pause to get the next snapshot from om
+    Thread.sleep(pauseInterval);
+
+    // HTTP call to /api/containers
+    String containerResponse = makeHttpCall(containerKeyServiceURL);
+    Map map = new Gson().fromJson(containerResponse, HashMap.class);
+    LinkedTreeMap linkedTreeMap = (LinkedTreeMap) map.get("data");
+    long reconMetadataKeyCount = (long)(double)linkedTreeMap.get("totalCount");
+
+    //verify count of keys after full snapshot
+    Assert.assertEquals(omMetadataKeyCount, reconMetadataKeyCount);
+
+    //verify if Recon Metadata captures vol0/bucket0/key0 info in container0
+    ArrayList containers = (ArrayList) linkedTreeMap.get("containers");
+    LinkedTreeMap nestedContainerMap = (LinkedTreeMap)containers.get(0);
+    Assert.assertEquals(1.0, nestedContainerMap.get("NumberOfKeys"));
```

Comment:
Can we also verify containerId if 0?

Reply:
If we don't verify containerId being what we expect it to be, we are not testing the public ContainerId -> Key mapping functionality in Recon.